### ISSUE TITLE
Developer laptop deployment enhancements

### ISSUE THREAD

**Issue by lbroudoux** [microcks/microcks#515]
We'd like to make the deployment of Microcks a breeze on developer's laptop !

As of today, the deployment of Microcks is a bit greedy and requires some resources on your laptop. We've demonstrated with https://microcks.io/blog/async-features-with-docker-compose/ that deploying the whole stack on a laptop is possible.

We've also heard from community feedback that the "1 Microcks par developer laptop" deployment pattern is a reality: small teams may not have the resources to deploy a mutualized instance at project startup. More over, we've heard of some people deploying Microcks in CI/CD pipeline - in a full automated way - without caring at all of all the Keycloak related stuffs... (see #484).

So for now we identified 2 enhancements to make using Microcks experience leaner on a developer laptop:
* Make the async part lighter trying to replace the [Strimzi](https://strimzi.io) Kafka distribution with [Red Panda](https://vectorized.io/redpanda). This one will be tracked by #516,
* Make the infrastructure part lighter removing Keycloak in a "developer mode" where user will have all administrator privileges. This one will be tracked by #484.

Obviously, we're open to any other suggestion! Let's start the discussion!

**Comment by lbroudoux** [comment]
Did some tests today, comparing previous docker-compose version with the new one with Red Panda and no Keycloak. Here are the results below:

```sh
$ docker-compose -f docker-compose.yml -f docker-compose-async-addon.yml up -d

$ docker stats
CONTAINER ID   NAME                       CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O       PIDS
3ded2034fa29   microcks-async-minion      2.30%     234.7MiB / 6.791GiB   3.38%     73.2kB / 55.3kB   4.1kB / 0B      72
0c5181a40525   microcks                   0.42%     311MiB / 6.791GiB     4.47%     114kB / 3.39MB    16.4kB / 86kB   46
4e202a2e93d0   microcks-kafka             3.55%     397.7MiB / 6.791GiB   5.72%     139kB / 103kB     0B / 328kB      84
e21006681ca8   microcks-db                0.91%     132.5MiB / 6.791GiB   1.91%     32.6kB / 19.6kB   0B / 0B         24
a1e8cdaea981   microcks-sso               1.69%     556.5MiB / 6.791GiB   8.00%     34kB / 504kB      4.1kB / 131kB   132
1e7d205bb125   microcks-postman-runtime   0.00%     34.51MiB / 6.791GiB   0.50%     24kB / 0B         0B / 0B         7
ebbc625f2f8b   microcks-zookeeper         0.42%     73.88MiB / 6.791GiB   1.06%     86.3kB / 59.3kB   0B / 434kB      52
```

With this configuration, Keycloak + ZK + Kafka = 1 GiB memory. The startup time is more or less 35 seconds on my 2 years old Mac Book Pro.

```sh
$ docker-compose -f docker-compose-devmode.yml up -d 

$ docker stats
CONTAINER ID   NAME                       CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O        PIDS
45ac8a43672f   microcks-async-minion      4.78%     229.3MiB / 6.791GiB   3.30%     28.1kB / 21kB     0B / 0B          72
0cf357b2ba0a   microcks                   0.48%     273MiB / 6.791GiB     3.93%     35.6kB / 32.8kB   0B / 86kB        46
63e1042d4617   microcks-kafka             25.98%    129.1MiB / 6.791GiB   1.86%     39.7kB / 27.8kB   24.6kB / 295kB   16
f579be98bca4   microcks-postman-runtime   0.00%     31.77MiB / 6.791GiB   0.46%     1.65kB / 0B       0B / 0B          7
6d0195c07cc7   microcks-db                0.67%     132.3MiB / 6.791GiB   1.90%     8.84kB / 17.5kB   0B / 0B          25
```

With this new configuration, we dropped ZK and Keycloak and Kafka usage is only 130 MiB. The startup time is more of less 20 seconds on the same 2 years old Mac Book Pro.

So basically, it would mean that we saved ~ 850 MiB of memory and reduced startup time of ~ 30%. Not too bad 😉 

What do you think? DO you have any other enhancements ideas?

**Comment by lbroudoux** [comment]
No more proposition for `1.5.0` release so we close this issue for now. Feel free to open a new one for other future enhancements !
